pkg/internal/resources/apiserver/apiserver.go

```diff
@@ -125,100 +124,105 @@ func Manifests(c Config) ([]unstructured.Unstructured, error) {
 			},
 		},
 	}
-	if c.Spec.StaticUsers != nil {
-		containers := deploymentPatch.Spec.Template.Spec.Containers
-		for i, container := range containers {
-			if container.Name == "manager" {
-				containers[i].Args = append(containers[i].Args,
-					"--htpasswd-secret-name=$(HTPASSWD_SECRET_NAME)",
-				)
-				containers[i].Env = append(containers[i].Env,
-					corev1.EnvVar{
-						Name:  "HTPASSWD_SECRET_NAME",
-						Value: c.Spec.StaticUsers.HtpasswdSecret.Name,
-					},
-				)
-				supportedAuth = append(supportedAuth, "Htpasswd")
+	var supportedAuth []string
+	for _, auth := range c.Spec.Authentication {
```

Comment:
```suggestion
	var supportedAuth []string
	present := make(map[string]struct{})
	for i, auth := range c.Spec.Authentication {
		if auth.StaticUsers != nil {
			if _, ok := present["statik"]; ok {
				return nil, fmt.Errorf("statik config redefined")
			}
			present["statik"] = struct{}{}
			if err := addStatikConfig(auth.StaticUsers, &deploymentPatch.Spec.Template.Spec); err != nil {
				return nil, err
			}
			supportedAuth = append(supportedAuth, "Statik")
		}
		if len(supportedAuth) != (i + 1) {
			return nil, fmt.Errorf("only a single auth provider can be defined per statement. got %v", supportedAuth[i + 1:])
		}
		...
	}
```

perhaps extract each auth provider to its own method; it's easier to read and follow the main function. Also, you're missing the checks if only a single one is non-nil.

Reply:
>Also, you're missing the checks if only a single one is non-nil.

we have this check in the `kubecarrier_webhook`

Reply:
defense in depth.

Reply:
add we have `if-else` here, so only the first one will be added